angr/procedures/libc/malloc.py

```diff
@@ -14,14 +14,4 @@ class malloc(angr.SimProcedure):
     def run(self, sim_size):
         self.argument_types = {0: SimTypeLength(self.state.arch)}
         self.return_type = self.ty_ptr(SimTypeTop(sim_size))
-
-        if self.state.solver.symbolic(sim_size):
-            size = self.state.solver.max_int(sim_size)
-            if size > self.state.libc.max_variable_size:
-                size = self.state.libc.max_variable_size
-        else:
-            size = self.state.solver.eval(sim_size)
-
-        addr = self.state.libc.heap_location
-        self.state.libc.heap_location += size
-        return addr
+        return self.state.heap._malloc(sim_size)
```

Comment:
Why should this simprocedure call `_malloc` instead of `malloc`? What's an example of a plugin that would not export a `malloc`?

Reply:
I see that the brk heap doesn't have a malloc, but instead has an allocate... I worry about the interface fragmentation, but I'm tentatively okay with this for now.

Reply:
My expectation is that most (if not all) plugins will export `malloc`, but yes, I was concerned about making it and `free` available publicly for the default heap since that one might not behave at all in the way a casual user would expect. On the other hand, it's just exposing the old functionality of the `SimProcedure`s, so maybe that isn't as big a deal as I had considered it.